### PR TITLE
Add support for URL parameters

### PR DESCRIPTION
**Pull request by lohmander**
As briefly discussed in #10. 

I didn't actually implement regex support, but rather plain `:param` style parameters. I'm passing the values as the third argument to the handler function. Maybe a better option would be to pass a request object, including both the body, params, headers, etc.

### CODE REVIEW

File: src/routing.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import {chain, merge, toPairs, trim} from 'lodash';
+
+const isParam = segment => segment.substr(0, 1) === ':';
```
Comment (muuki88) [inline src/routing.js]: I think `startsWith` is clearer than `substr(0,1`.

File: src/routing.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+const toSegments = pathname => trim(pathname, '/').split('/');
+
+const buildRoute = (segments, methods) => {
```
Comment (muuki88) [inline src/routing.js]: Can you add some documentation to this method, e.g.

- Short explanation what this method does
- What types should the input parameters have
- What's the return type

File: src/routing.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  const segment = segments.shift();
+  const edge = segments.length === 0;
```
Comment (muuki88) [inline src/routing.js]: I would rather use `isEdge` to indicate that this is a bool value.
`isLastSegment` is the better name here ( as you used below )

File: src/routing.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import {chain, merge, toPairs, trim} from 'lodash';
+
+const isParam = segment => segment.substr(0, 1) === ':';
+const toSegments = pathname => trim(pathname, '/').split('/');
+
+const buildRoute = (segments, methods) => {
+  const segment = segments.shift();
```
Comment (muuki88) [inline src/routing.js]: `shift` mutates the `segments` array, right? I think we should keep these mutation / side effects out of a function.
Try using the [array spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) instead, e.g.

```js
const [nextSegment, ...remainingSegments] = segments;
const isLastSegment = !(remainingSegments && remainingSegments.length > 0);
```

File: src/routing.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    )
+  };
+};
+
+const buildParams = (param, value) =>
+  param ? { params: { [param.substr(1)]: value } } : {};
+
+const lookup = (segments, routes) => {
+  const segment = segments.shift();
+  const isLastSegment = segments.length === 0;
```
Comment (muuki88) [inline src/routing.js]: The same as above. Use array spreading instead of shift to avoid mutation.

File: src/routing.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        ? { $edge: edge, $methods: methods }
+        : buildRoute(segments, methods)
+    )
+  };
+};
+
+const buildParams = (param, value) =>
+  param ? { params: { [param.substr(1)]: value } } : {};
+
+const lookup = (segments, routes) => {
```
Comment (muuki88) [inline src/routing.js]: Please add some documentation for this method. What are `segments` and `routes` types and what are the expected return types.

**Review by muuki88 — CHANGES_REQUESTED** [review]
Thanks a lot for your good work on this @lohmander :heart: :smiley: 

I made a few comments, that should be trivial to fix or be discussed about :)

Very well done pull request with description, tests and updated documentation :+1:

**Comment by lohmander** [comment]
All sound points, I'll look into fixing everything next week. 👍

**Comment by lohmander** [comment]
@muuki88 I think all of the issues should be fixed now.

**Review by muuki88 — APPROVED** [review]
Awesome! Thanks a lot. I'll release this ASAP

**Pull request merged**